db/import.go

```diff
@@ -298,11 +298,17 @@ func (db *Database) importDoc(docid string, body Body, isDelete bool, existingDo
 
 // Migrates document metadata from document body to system xattr.  On CAS failure, retrieves current doc body and retries
 // migration if _sync property exists.  If _sync property is not found, returns doc and sets requiresImport to true
-func (db *Database) migrateMetadata(docid string, body Body, existingDoc *sgbucket.BucketDocument) (docOut *Document, requiresImport bool, err error) {
+func (db *Database) migrateMetadata(docid string, body Body, existingDoc *sgbucket.BucketDocument, requiresMigrationDocFlag bool) (docOut *Document, requiresImport bool, err error) {
 
-	// Unmarshal the existing doc in legacy SG format
-	doc, unmarshalErr := unmarshalDocument(docid, existingDoc.Body)
-	if err != nil {
+	var doc *Document
+	var unmarshalErr error
+
+	if requiresMigrationDocFlag {
```

Comment:
Change to if existingDoc.Xattr != nil, and add a comment describing the scenario this is covering.